docs/develop/gettingstarted.rst

```diff
@@ -0,0 +1,310 @@
+If it built successfully, you should see a message similar to this:
+
+.. code-block:: console
+
+   LINK mpy-cross
+      text	   data	    bss	    dec	    hex	filename
+    279328	    776	    880	 280984	  44998	mpy-cross
+
+.. note::
+
+   Use ``make -C mpy-cross`` to build the cross-compiler in one statement
+   otherwise, you will need to do ``cd ..`` for the next step.
+
+Build for the Unix Port
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To build for the unix port, make sure all linux related dependencies are 
+installed as detailed in the required dependencies section. Also, make sure you have a 
+working environment for ``gcc`` and ``GNU make``. Ubuntu 20.04 has been used for the example
+below but other unixes ought to work with little modification:
+
+.. code-block:: console
+
+   $ gcc --version
+   gcc (Ubuntu 9.3.0-10ubuntu2) 9.3.0
+   Copyright (C) 2019 Free Software Foundation, Inc.
+   This is free software; see the source for copying conditions.  There is NO
+   warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.then build:
+
+.. code-block:: console
+
+   $ cd ports/unix
+   $ make submodules
+   $ make
+
+If MicroPython built correctly, you should see the following:
+
+.. code-block:: console
+
+   LINK micropython
+      text	   data	    bss	    dec	    hex	filename
+    412033	   5680	   2496	 420209	  66971	micropython
+
+Now run it:
+
+.. code-block:: console
+
+   $ ./micropython
+   MicroPython v1.13-38-gc67012d-dirty on 2020-09-13; linux version
+   Use Ctrl-D to exit, Ctrl-E for paste mode
+   >>> print("hello world")
+   hello world
+   >>>
+
+Build for the stm32 port
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Like the unix port, you need to install some required dependencies
+as detailed in the required dependencies section, then build:
+
+.. code-block:: console
+
+   $ cd ports/stm32
+   $ make submodules
+   $ make
+
+Please refer to the `stm32 documentation <https://github.com/micropython/micropython/tree/master/ports/stm32>`_ 
+for more details on flashing the firmare.
+
+Build for the Windows port
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Windows port is experimental and based on the Unix port. Build on 
+debian/linux systems using the following commands:
+
+.. code-block:: console
+
+   $ cd ports/windows
+   $ sudo apt-get install gcc-mingw-w64
+   $ make CROSS_COMPILE=i686-w64-mingw32-
+
+Then run easily with ``wine``:
+
+.. code-block:: console
+
+   $ wineconsole --backend=curses ./micropython.exe
+
+Refer to the `windows port documentation <https://github.com/micropython/micropython/tree/master/ports/windows>`_
+on more details on building using Cygwin and MS visual Studio 2013 or higher.
+
+Build the documentation
+------------------------
+
+MicroPython documentation is created using ``Sphinx``. If you have already
+installed Python, then install ``Sphinx`` using ``pip``:
+
+.. code-block:: console
+
+   $ python3 -m pip install sphinx
```

Comment:
I don't know what to recommend here, but I feel that we should be encouraging people to use a virtual environment for installing Python packages...